### PR TITLE
Platforms2

### PR DESCRIPTION
**Pull request by trzy**
Add QueryTopology_FindLargestPositionsSittable() to Unity and update the Spatial Understanding demo with this query. This is the companion to: Microsoft/HoloToolkit#86

Note: this should replace Platforms #651.

### CODE REVIEW

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/Scripts/UI.cs [hunk 1]
```diff
@@ -221,6 +221,7 @@ private void SetupMenus()
             AddButton("Large positions on floor", Panels.Topology, () => { SpaceVisualizer.Instance.Query_Topology_FindLargestPositionsOnFloor(); timeLastQuery = DateTime.MinValue; });
             AddButton("Place objects positions", Panels.Topology, () => { SpaceVisualizer.Instance.Query_Topology_FindPositionsPlaceable(); timeLastQuery = DateTime.MinValue; });
 
+
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/Scripts/UI.cs]: nit: extra line?

**Comment by trzy** [comment]
Extra line should be fixed now.

**Comment by NeerajW** [comment]
Just to help devs understand, what do you mean by sittable area?

**Comment by trzy** [comment]
That's a good question :) This is the terminology used by the 
SpatialUnderstanding library itself and is not something I came up with. 
My understanding is that these are locations on top of platforms 
(spanning the entire area of the platform, unlike "platform edge" 
queries).  I've updated the spatial understanding demo as well so you 
can try out this query and see the kinds of results it returns.  I use 
it my games to place things on tables, couches, beds, etc. (for example, 
vehicles on the bed and couch in this demo were placed using this query: 
https://www.youtube.com/watch?v=Ltr9uVWC-5w )

**Comment by jessemcculloch** [comment]
For clarification, like @trzy mentioned, the term sittable comes from the HoloToolkit/SpatialUnderstanding/Topology queries and is defined as a horizontal plane between the Min and Max heights that conforms to the specified surface area.  It could be a chair, it could be a couch, it could be many other things, as long as the surface area of the plane matches, as well as the height falls between the two values.

**Comment by keveleigh** [comment]
Is this change dependent on a new DLL drop?

**Comment by StephenHodgson** [comment]
That is correct

**Comment by jessemcculloch** [comment]
Why does it need a new dll drop? 
(Not arguing, just not aware of what causes that need)

**Comment by keveleigh** [comment]
The change on the native side of things, [here](https://github.com/Microsoft/MixedRealityToolkit/pull/86/files), added the new method that's [imported in this PR](https://github.com/trzy/HoloToolkit-Unity-trzy/blob/81e120ba26eb8c529036fdb9670cb55775b9a6a3/Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingDllTopology.cs#L134). Without it, there should be some sort of import failure.

**Comment by jessemcculloch** [comment]
That's odd, I tested it with no errors at all

**Comment by StephenHodgson** [comment]
Only blocking this until we get the binaries updated.

**Comment by trzy** [comment]
Anything needed from me? Any ETA?

**Comment by StephenHodgson** [comment]
Nope, you're good. Still waiting on MSFT to sign the binaries.

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/Scripts/UI.cs [hunk 2]
```diff
@@ -247,6 +248,7 @@ private void SetupMenus()
             AddButton("OnEdge NearCenter", Panels.LevelSolver, () => { LevelSolver.Instance.Query_OnEdge_NearCenter(); timeLastQuery = DateTime.MinValue; });
             AddButton("OnFloor AwayFromMe", Panels.LevelSolver, () => { LevelSolver.Instance.Query_OnFloor_AwayFromMe(); timeLastQuery = DateTime.MinValue; });
             AddButton("OnFloor NearMe", Panels.LevelSolver, () => { LevelSolver.Instance.Query_OnFloor_NearMe(); timeLastQuery = DateTime.MinValue; });
+            AddButton("Large positions sittable", Panels.Topology, () => { SpaceVisualizer.Instance.Query_Topology_FindLargePositionsSittable(); timeLastQuery = DateTime.MinValue; });
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/Scripts/UI.cs]: This creates a duplicate button in the Topology menu. Was in meant to go under `Panels.LevelSolver`, like the other buttons near it in this list?

Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/SpatialUnderstanding/Scripts/UI.cs]: @trzy

**Comment by trzy** [comment]
I guess I must have mistakenly added that button in two different places 
-- Panels.Topology and Panels.LevelSolver.  It should only exist for 
Panels.Topology.  Please remove the redundant line that is grouped with 
the LevelSolver panel calls.  I would do this myself but I'm frozen back 
in HoloToolkit-Unity still.

**Pull request merged**